### ISSUE TITLE
Feature request: Alternate approach to numeric quick jump to access beyond link 10

### ISSUE THREAD

**Issue by pixdrift** [makeworld-the-better-one/amfora#131]
Thanks for all your work @makeworld-the-better-one!

I really like the numeric quick jump feature using numeric selection (0-9) to access links, but I am noticing as Gemini pages get larger, the number of pages that have more than 10 links is more frequent, which results in the fallback of using [tab] to access links, which I have found can cause issues with scrolling as it shifts focus.

**Approach 1:**
Would it be possible to add a 'dwell' or 'timeout' when the first number is pressed if there is more than 10 links on the page to determine of the user is attempting to access a higher number eg. link 45? If a second number isn't pressed in this time, fall back to the single number link.

**Approach 2:**
Would it be possible to tweak the approach slightly to allow accessing links past 10 by left padding links with 0, and then only jumping when the complete number is entered? eg.

For pages with <10 links, retain current behaviour (except remove 0)
```
[1]  This is a link
...
[9]  This is another link
```

For pages with >9 links, determine the total number of links on the page and left pad, eg. if the page has 45 links on it, pad all links to 2 digits eg.
```
[01]  This is a link
...
[45] This is another link
```

In the scenario where there are >9 links and < 100 links, wait for the user to type 2 numbers before quick jumping. To access link '1' the behaviour changes to typing '0', '1' to trigger the quick jump.

If the page has >99 links, renumber all links with 3 numbers and make the input wait for 3 numbers before jumping, etc.

The downside to this approach is that the total number of links on the page would need to be determined before rendering. I haven't looked at the source to see if this behaviour would align with the current rendering approach.

**Comment by makeworld-the-better-one** [comment]
Thanks for your interest in Amfora!

> I really like the numeric quick jump feature using numeric selection (0-9) to access links, but I am noticing as Gemini pages get larger, the number of pages that have more than 10 links is more frequent, which results in the fallback of using [tab] to access links, which I have found can cause issues with scrolling as it shifts focus.

You can access links beyond 10 easily, by pressing the spacebar, typing the link number, and pressing enter. This is fast enough for most cases in my experience. This is documented on the help page, but you are about the third or fourth person to not realize this (but then also report it! There must be many others), so it clearly needs some emphasis. I plan on documenting basic navigation, among other features, in a wiki, as described in #124.

As for the approaches you propose, I think they're not bad, but that the current solution is the most elegant and consistent one. It works quickly and in the same way for every page. Now that you know it exists, what do you think? When you try it out, is it easy for you to use?

**Comment by pixdrift** [comment]
Apologies @makeworld-the-better-one I did miss this in the documentation. I looked over the help page (?), but missed the 'spacebar' reference, I think because it is sorted by hot key rather than the function ie. I was looking for 'navigation' functions. But as you have suggested this looks to be changing in the documentation.

Space bar is definitely easy enough, and appreciate the functionality. Cheers!
